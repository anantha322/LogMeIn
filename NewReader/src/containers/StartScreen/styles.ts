import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  highlight: {
    fontWeight: '700',
  },
  titleText: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  container: {
    backgroundColor: 'white',
    flex: 1,
  },
  emptyList: {flex: 1, alignItems: 'center', justifyContent: 'center'},
});
