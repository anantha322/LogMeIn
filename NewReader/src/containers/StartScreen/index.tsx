import {get} from 'lodash';
import React, {useEffect} from 'react';
import {FlatList, Text, View} from 'react-native';
import {useSelector, useDispatch} from 'react-redux';

import {
  getRssNewsRequest,
  getRssNewsSelector,
} from '../../adapters/redux/rssReader';
import {NewsListItem} from '../../components';
import {styles} from './styles';

const StartScreen = (props: any) => {
  const dispatch = useDispatch();
  const newsSelector = useSelector(getRssNewsSelector);

  useEffect(() => {
    dispatch(getRssNewsRequest());
  }, []);

  const onPressItem = (item: any) => {
    props.navigation.navigate('Details', {selectedNewsFeed: item});
  };

  const renderItem = ({item}: any) => {
    if (item && item.id && item.title) {
      return (
        <NewsListItem
          id={item?.id}
          title={item?.title}
          onPressItem={() => onPressItem(item)}
        />
      );
    }
    return <></>;
  };
  const renderEmptyComponent = () => {
    return (
      <View style={styles.emptyList}>
        <Text>No feed available</Text>
      </View>
    );
  };
  return (
    <View style={styles.container}>
      <FlatList
        data={newsSelector}
        renderItem={renderItem}
        keyExtractor={(item: any) => get(item, 'id', 0)}
        ListEmptyComponent={renderEmptyComponent()}
      />
    </View>
  );
};

export default StartScreen;
