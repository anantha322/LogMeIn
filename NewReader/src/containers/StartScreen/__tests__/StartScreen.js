import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});
import StartScreen from '../';

const props = {};
const initialState = {};
const mockStore = configureStore();
let store;

describe('render Detail screen', () => {
  test('renders correctly', () => {
    store = mockStore(initialState);

    const tree = renderer
      .create(
        <Provider store={store}>
          <StartScreen {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
