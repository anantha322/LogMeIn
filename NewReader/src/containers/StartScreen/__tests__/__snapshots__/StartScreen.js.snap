// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Detail screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Component>
          No feed available
        </Component>
      </Component>
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text>
          No feed available
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
